backend/app/Repositories/EloquentChartVisitRepository.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Repositories;
+
+use App\Contracts\Visits\PageViewsFilterData;
+use App\DataTransformer\DataTransformerInterface;
+use App\DataTransformer\visits\ChartVisitDataTransformer;
+use App\Repositories\Contracts\ChartVisitRepository;
+use Illuminate\Support\Facades\DB;
+
+final class EloquentChartVisitRepository implements ChartVisitRepository
+{
+    public function findByFilter(PageViewsFilterData $filterData, int $interval): DataTransformerInterface
```

Comment:
After you get rid of interface PageViewsFilterData, use here DatePeriod instead of PageViewsFilterData as type hint